### ISSUE TITLE
local-path-storage package delete fails when configuration is: `namespace: default`

### ISSUE THREAD

**Issue by ciberkleid** [vmware-tanzu/community-edition#3067]
## Bug Report

If local-path-storage package is installed with `namespace: default`, then the following error occurs when deleting the package:

```
$ tanzu package installed delete local-path-storage
Deleting installed package 'local-path-storage' in namespace 'default'. Are you sure? [y/N]: y
| Uninstalling package 'local-path-storage' from namespace 'default' 
| Getting package install for 'local-path-storage' 
/ Deleting package install 'local-path-storage' from namespace 'default' 


Error: resource deletion failed: kapp: Error: Applying delete namespace/default (v1) cluster:
  Deleting resource namespace/default (v1) cluster: namespaces "default" is forbidden: this namespace may not be deleted (reason: Forbidden)
. Delete failed: Error (see .status.usefulErrorMessage for details)
Error: exit status 1

✖  exit status 1 
```

As a result, the package is not deleted:

```
$ tanzu package installed list
| Retrieving installed packages... 
  NAME                PACKAGE-NAME                                   PACKAGE-VERSION  STATUS                                                             
  local-path-storage  local-path-storage.community.tanzu.vmware.com  0.0.20           Delete failed: Error (see .status.usefulErrorMessage for details)  
```

and:

```
$ tanzu package installed get local-path-storage
| Retrieving installation details for local-path-storage... 
NAME:                    local-path-storage
PACKAGE-NAME:            local-path-storage.community.tanzu.vmware.com
PACKAGE-VERSION:         0.0.20
STATUS:                  Delete failed: Error (see .status.usefulErrorMessage for details)
CONDITIONS:              [{DeleteFailed True  Error (see .status.usefulErrorMessage for details)}]
USEFUL-ERROR-MESSAGE:    kapp: Error: Applying delete namespace/default (v1) cluster:
  Deleting resource namespace/default (v1) cluster: namespaces "default" is forbidden: this namespace may not be deleted (reason: Forbidden)
```

## Expected Behavior

The package should be deleted.
There should not be an attempt to delete the default namespace.

## Steps to Reproduce the Bug

Create a local-path-storage-values.yaml files with contents:
```
namespace: default
```

Install local-path-storage package using this values file, then delete the package.

**Comment by ciberkleid** [comment]
I think actually this is a kapp issue, not a Tanzu Community Edition issue... In general, kapp should probably have a safeguard whereby it does not try to delete the default namespace and still enables successful deletion of the thing that is being deleted.

Should I close this issue here or is there a way to reassign this or create a linked issue or something like that?

**Comment by stmcginnis** [comment]
Let's keep this open. If you do open an issue with kapp, if you would be able to add a link to that here, then we could use this as a tracking issue to make sure that change makes it all the way through to a TCE release.

Thanks @ciberkleid!

**Comment by ciberkleid** [comment]
Logged it under kapp:

[https://github.com/vmware-tanzu/carvel-kapp/issues/452](https://github.com/vmware-tanzu/carvel-kapp/issues/452)

**Comment by joshrosso** [comment]
Thanks for following up on this @ciberkleid!

**Comment by 100mik** [comment]
The fix for this has gone in with kapp v0.47.0.
We now use the "orphan" delete strategy which removes ownership labels for some resources like the `default` ns.

This should be resolved when `kapp-controller` picks it up.
(cc: @neil-hickey maybe you could drop a comment here when kapp-controller picks it up?)
